actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/tac/UserTnCActor.java

```diff
@@ -59,13 +62,26 @@ private void acceptTNC(Request request) {
       isManagedUser = true;
     }
     SystemSettingClient systemSettingClient = SystemSettingClientImpl.getInstance();
-    String latestTnC =
-        systemSettingClient.getSystemSettingByFieldAndKey(
-            getActorRef(ActorOperations.GET_SYSTEM_SETTING.getValue()),
-            JsonKey.TNC_CONFIG,
-            JsonKey.LATEST_VERSION,
-            new TypeReference<String>() {},
-            context);
+    String tncType = (String) request.getRequest().get(JsonKey.TNC_TYPE);
+    String latestTnC = "";
+    // if tncType is null , continue to use the same field for user tnc acceptance
+    if (StringUtils.isNotBlank(tncType)) {
```

Comment:
If only string is changing, then only that must be if-else clause. The function needn't be repeated.

`if (StringUtils.isBlank(tncType)) {
   tncType = JsonKey.TNC_CONFIG; // default
}
String latestTnC = systemSetting call
`

Reply:
updated.